src/v/cluster/members_backend.cc

```diff
@@ -467,14 +467,28 @@ size_t members_backend::calculate_total_replicas(
  **/
 members_backend::unevenness_error_info
 members_backend::calculate_unevenness_error(
-  partition_allocation_domain domain) const {
+  const update_meta& update, partition_allocation_domain domain) const {
     static const std::vector<partition_allocation_domain> domains{
       partition_allocation_domains::consumer_offsets,
       partition_allocation_domains::common};
 
     const auto node_cnt = _allocator.local().state().available_nodes();
 
-    const auto node_replicas = calculate_replicas_per_node(domain);
+    auto node_replicas = calculate_replicas_per_node(domain);
+    /**
+     * adjust per node replicas with the replicas that are going to be removed
+     * from the node after successful reallocation
+     */
+    for (const auto& r : update.partition_reallocations) {
+        if (r.allocation_units) {
+            for (const auto& to_remove : r.replicas_to_remove) {
```

Comment:
I wonder, is it guaranteed that the only replicas that are present in the old replica set but not in the new one are `replicas_to_remove`?

Reply:
yes, this is guaranteed, moreover currently replicas to remove contain only single node